src/data/fin/basic.lean

```diff
@@ -260,6 +260,10 @@ by simp [lt_iff_coe_lt_coe]
 lemma eq_last_of_not_lt {i : fin (n+1)} (h : ¬ (i : ℕ) < n) : i = last n :=
 le_antisymm (le_last i) (not_lt.1 h)
 
+lemma last_eq_top (n : ℕ) : fin.last n = ⊤ := rfl
+
+lemma zero_eq_bot (n : ℕ) : (0 : fin (n + 1)) = ⊥ := rfl
```

Comment:
Also, you should be able to use `zero_eq_bot` directly rather than reproving it for `fin`.

Reply:
There is [`bot_eq_zero`](https://leanprover-community.github.io/mathlib_docs/algebra/order/monoid.html#bot_eq_zero), but it need `canonically_ordered_add_monoid` so it doesn't work for `fin (n + 1)`. But it probable makes sense to state them the other way.

Reply:
Argh, `fin` isn't currently a `canonically_linear_ordered_add_monoid` because addition is looping around, not saturating. I always forget.